src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/InProcDataCollectionExtensionManager.cs

```diff
@@ -45,10 +54,23 @@ internal class InProcDataCollectionExtensionManager
         /// The default codebase to be used by inproc data collector
         /// </param>
         public InProcDataCollectionExtensionManager(string runSettings, ITestEventsPublisher testEventsPublisher, string defaultCodeBase)
+            : this(runSettings, testEventsPublisher, defaultCodeBase, new FileHelper())
+        {}
+
+        protected InProcDataCollectionExtensionManager(string runSettings, ITestEventsPublisher testEventsPublisher, string defaultCodeBase, IFileHelper fileHelper)
         {
             this.InProcDataCollectors = new Dictionary<string, IInProcDataCollector>();
             this.inProcDataCollectionSink = new InProcDataCollectionSink();
             this.defaultCodeBase = defaultCodeBase;
+            this.fileHelper = fileHelper;
+            this.codeBasePaths = new List<string> { this.defaultCodeBase };
+
+            // Get Datacollector codebase paths from test plugin cache
+            var extensionPaths = TestPluginCache.Instance.GetExtensionPaths(DataCollectorEndsWithPattern);
```

Comment:
>extensionPaths [](start = 16, length = 14)

can we look at passing this in the constructor and not call TestPluginCache from here..?

Reply:
Done.